### ISSUE TITLE
About BatchNorm QAT 's problems

### ISSUE THREAD

**Issue by typhoonlee** [microsoft/nni#5704]
**Describe the bug**:
when i use QAT_Quantizer() to qat my model，i got a wrap_model.pt，then i test with this model，encounter this bug：
![image](https://github.com/microsoft/nni/assets/43179227/5db45961-4e4e-45f4-8dcf-3102f0ec0c70)

**Environment**:  linux
- NNI version:   __version__ = '999.dev0'
- Training service (local|remote|pai|aml|etc):   remote
- Python version:   3.7.10
- PyTorch version:  1.12.0 + cu113
- Cpu or cuda version:  GPU version GeForce RTX 3090

**Reproduce the problem**
- Code|Example:
![image](https://github.com/microsoft/nni/assets/43179227/2a318a81-b51f-4233-b0c3-fb83089f9a8f)
![image](https://github.com/microsoft/nni/assets/43179227/36b2f958-f90a-485f-b27f-9ada7421123a)

and my model arch is :
![image](https://github.com/microsoft/nni/assets/43179227/c512391c-aa74-46bb-a42a-831497ac41d5)
![image](https://github.com/microsoft/nni/assets/43179227/09b7eb6c-401c-43b7-aced-dbc04418f2f9)

- How to reproduce:
